packages/plugin-degenfans-alfafrens/src/actions/stakingConsultantAction.ts

```diff
@@ -0,0 +1,133 @@
+export const stakingConsultantAction: Action = {
+    name: "GET_ALFAFRENS_STAKING_RECOMENDATION",
+    similes: [
+        "VIEW_ALFAFRENS_STAKING_RECOMENDATION"
+    ],
+    description: "get recomendation for AlfaFrens stakings",
+    suppressInitialMessage: true,
+    validate: async () => true,
+    handler: async (
+        runtime: IAgentRuntime,
+        message: Memory,
+        state: State,
+        _options: { [key: string]: unknown },
+        callback: HandlerCallback
+    ) => {
+        try {
+            elizaLogger.log("Starting GET_ALFAFRENS_STAKING_RECOMENDATION handler...");
+
+            // Initialize or update state
+            if (!state) {
+                state = (await runtime.composeState(message)) as State;
+            } else {
+                state = await runtime.updateRecentMessageState(state);
+            }
+
+            const moxieUserInfo: MoxieUser = state.moxieUserInfo as MoxieUser;
+
+            const context = composeContext({
+                state,
+                template: stakingConsultantTemplate,
+            });
+
+            const transferDetails = await generateObject({
+                runtime,
+                context,
+                modelClass: ModelClass.SMALL,
+                schema: StakingSchema,
+            });
+            let {
+                userAddress,
+                amount,
+                mysubs,
+                mystake,
+                minsubs,
+            } = transferDetails.object as {
+                amount: number;
+                userAddress: string;
+                mysubs: boolean;
+                mystake: boolean;
+                minsubs: number;
+            };
+
+            const userData = getUserData(moxieUserInfo);
+
+            const stakingData: StakingRequest = { amount: amount, mysubs: mysubs, mystake: mystake, minsubs: minsubs };
+            const resp = await getStakingOptions(userData, userAddress, stakingData);
+            let tbl: string = "";
+            console.log(resp);
+            if (resp.status == 200) {
+                tbl += "\n";
+                if (resp.data && resp.data.result && resp.data.result.stakingOptions && resp.data.result.stakingOptions.length > 0) {
+                    tbl += "|rank|AlfaFrens Channel|ROI Spark/mo|current stake|\n";
+                    tbl += "|------:|:--------|----:|------|\n";
+                    resp.data.result.stakingOptions.forEach(e => {
+                        tbl += "|#" + e.rank + "|[" + e.name + "|https://alfafrens.com/channel/" + e.channelAddress + "]|" + e.roi + "|" + e.currentStake + "|\n";
+                    });
+                } else {
+                    tbl += "no staking options found";
+                }
+
+                if (resp.data.result.amountRandom) {
+                    tbl += "\n* you can also specify a staking amount to get a more precise result, e.g. 15000 AF"
+                }
+                tbl += getHelpText(resp.data.user);
```

Comment:
I recommend that you put this `getHelpText` on the top, so the first thing user see is if the skill was able to find the associated alfafrens account or not + prompt for sign up

Then provide recommendation for that account or a generic recommendation if there's no account

A bit confusing when the order is reversed 🤔 since I was given recommendation and thought it was personalized but then said no account at the end